tests/integration/query/inline_array/with_filter_all_test.go

```diff
@@ -0,0 +1,305 @@
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package inline_array
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestQueryInlineStringArrayWithAllFilter(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple inline array, filtered all of string array",
+		Actions: []any{
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Shahzad",
+					"pageHeaders": ["first", "second"]
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Fred",
+					"pageHeaders": [null, "second"]
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(filter: {pageHeaders: {_all: {_ne: null}}}) {
+						name
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"name": "Shahzad",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	executeTestCase(t, test)
+}
+
+func TestQueryInlineNonNullStringArrayWithAllFilter(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple inline array, filtered all of non null string array",
+		Actions: []any{
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Shahzad",
+					"preferredStrings": ["first", "second"]
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Fred",
+					"preferredStrings": ["", "second"]
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(filter: {preferredStrings: {_all: {_ne: ""}}}) {
+						name
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"name": "Shahzad",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	executeTestCase(t, test)
+}
+
+func TestQueryInlineIntArrayWithAllFilter(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple inline array, filtered all of int array",
+		Actions: []any{
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Shahzad",
+					"testScores": [50, 80]
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Fred",
+					"testScores": [null, 60]
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(filter: {testScores: {_all: {_ne: null}}}) {
+						name
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"name": "Shahzad",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	executeTestCase(t, test)
+}
+
+func TestQueryInlineNonNullIntArrayWithAllFilter(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple inline array, filtered all of non null int array",
+		Actions: []any{
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Shahzad",
+					"testScores": [50, 80]
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "Fred",
+					"testScores": [0, 60]
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(filter: {testScores: {_all: {_lt: 70}}}) {
+						name
+					}
```

Comment:
suggestion: would be also great to see a reassurance (in for of another test) that this new op works with compound conditions. For example:
`filter: {testScores: {_all: {_and: [{_lt: 70}, {_gte: 0}]}}}`
And to do the same with other new ops.

Reply:
I've added two compound tests.